### ISSUE TITLE
release-it forgets about change in package.json

### ISSUE THREAD

**Issue by sebs** [webpro/release-it#279]
Using latest release ;) 

```shell
> release-it patch --non-interactive

✔ Init (release 0.5.2 for xxxxx)
✔ Bump version
⠹ Git commitWARNING No changes to commit. The latest commit will be tagged.
✔ Git commit
✔ Git tag
✔ Git push
✔ Done (in 6s.)
```

Intended result: package.json is mutated and pushed

Result: package.json is mutated, but not pushed. 

my config

```json
{
  "src": {
    "tagName": "v%s"
  },
  "github": {
    "release": false
  },
  "npm": {
    "publish": false
  }
}
```

**Comment by webpro** [comment]
Hi @sebs!

Do you have a reproducible test case?

My gut tells me the version in `package.json` was already at the next version. Could this be the case? Can you double-check the latest Git tag matches `package.json#version` with the next release?

**Comment by sebs** [comment]
it happens reproducably on each release. 
I resorted to add a postrelease step. 

release-it is triggered via a npm script

**Comment by sebs** [comment]
well, actually ... cant reproduce it with a empty repo

**Comment by IamTheHttp** [comment]
I think @webpro comment regarding the git tag is the correct approach.
I tried reproducing this issue as well, The only way I could reproduce it is by manually advancing the package.json version ahead, and then trying to release-it to that version, this creates the exact behaviour above.

I'm also curious about the output of git tag -l

**Comment by sebs** [comment]
output

```
git tag -l
.....
v0.0.5
v0.0.6
v0.0.7
v0.0.8
v0.0.9
v0.1.0
v0.1.1
v0.1.2
v0.1.3
v0.2.0
v0.2.1
v0.2.2
v0.2.3
v0.2.4
v0.2.5
v0.3.0
v0.3.1
v0.3.2
v0.3.3
v0.3.4
v0.3.5
v0.3.6
v0.3.7
v0.3.8
v0.4.0
v0.4.1
v0.4.2
v0.4.3
v0.4.4
v0.5.0
```

I swear (tm), I am not changing the version number myself

**Comment by webpro** [comment]
What happens internally, is that release-it reads the latest tag using `git describe --tags --abbrev=0`, and uses the `version` from package.json as a fallback. Then increments the result, e.g. with `patch` `v0.5.0` becomes `v0.5.1`. The `version` in package.json is updated, and this change is staged and committed. This commit is then tagged as `v0.5.1`.

An error (e.g. can't write file) bumping `package.json` should result in an error.

I've also checked that the `v` prefix might mess things up, which isn't the case.

Maybe a run using `--verbose` and `--debug` reveals something.

**Comment by IamTheHttp** [comment]
Can we maybe check if the latest tag does not match the package json version?
this doesn't have to be a blocking error, but a soft warning that the user will see.

"Warning, latest tag found was %tag, package.json version is %version, are you sure you want to continue?"

We also encountered a similar problem.

**Comment by sebs** [comment]
@webpro Oh, my bad for not giving the help a look. It might have to do with my prerelease step: A polymer selenium test. This might not error code (0) correctly enough. I will give it a try. 

Hey btw: Thank you for answering this!

**Comment by sebs** [comment]
Thing is: commitplease breaks the commit. 

Rightly so. I expect a 'Release: version` and not 'Release version' message. The would make the difference.

**Comment by webpro** [comment]
@sebs Not sure I understand. Sounds like we can close this ticket?

**Comment by webpro** [comment]
@IamTheHttp, @sebs, I've released [v7.4.0](https://github.com/webpro/release-it/releases/tag/7.4.0), which warns for inequality between latest git tag and npm version.

**Comment by sebs** [comment]
<3

Yes you can. The bug disappeared after a reinstall. Not sore what this was. But for sure the same message "Release 1.0.0" now goes through. Very much apprechiate your help. Thank you. Even if this was a heisenbug!
